### PR TITLE
MOD: reconstruct test code

### PR DESCRIPTION
**Pull request by xuanyu66**
- Split FlinkTest to multi test class
- Use Parameterized for multi-mode
- Use Rule to set up and tear down environment.

### CODE REVIEW

**Comment by xuanyu66** [comment]
@humengyu2012 @shiyuhang0 @zhangyangyu PTAL

**Comment by xuanyu66** [comment]
run-all-tests

**Comment by xuanyu66** [comment]
run-all-tests

**Review by zhangyangyu — APPROVED** [review]

File: flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkStreamingModeTest.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    private static final Set<Long> randomLong = new HashSet<>();
+
+    static {
+      while (randomLong.size() < 3) {
+        randomLong.add((long) RANDOM.nextInt(100));
+      }
+    }
+
+    public RandomExceptionMapFunction() {
+      System.out.println("----------------------------");
```
Comment (humengyu2012) [inline flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkStreamingModeTest.java:136]: Can we use log.info?

File: flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkStreamingModeTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    @Override
+    public RowData map(Long value) throws Exception {
+      GenericRowData rowData = new GenericRowData(1);
+      rowData.setField(0, value);
+      // Waiting for checkpoint
+      Thread.sleep(50L);
+      if (randomLong.remove(value)) {
+        System.out.println("----------------------------" + value);
+        throw new IllegalStateException();
```
Comment (humengyu2012) [inline flink/flink-1.14/src/test/java/io/tidb/bigdata/flink/tidb/sink/TiKVSinkStreamingModeTest.java:149]: Add a message for exception? such as "Test exception".

**Pull request merged**